src/plugins/data/common/search/tabify/response_writer.ts

```diff
@@ -65,9 +65,37 @@ export class TabbedAggResponseWriter {
     }
   }
 
-  response(): TabbedTable {
+  response(): Datatable {
     return {
-      columns: this.columns,
+      type: 'datatable',
+      columns: this.columns.map((column) => {
+        const cleanedColumn: DatatableColumn = {
+          id: column.id,
+          name: column.name,
+          meta: {
+            type: column.aggConfig.params.field?.type || 'number',
+            field: column.aggConfig.params.field?.name,
+            index: column.aggConfig.getIndexPattern()?.title,
+            params: column.aggConfig.toSerializedFieldFormat(),
+            source: 'esaggs',
+            sourceParams: {
+              indexPatternId: column.aggConfig.getIndexPattern()?.id,
+              appliedTimeRange:
+                column.aggConfig.params.field?.name &&
+                this.params.timeRange &&
+                this.params.timeRange.timeFields &&
+                this.params.timeRange.timeFields.includes(column.aggConfig.params.field?.name)
+                  ? {
+                      from: this.params.timeRange?.from?.toISOString(),
+                      to: this.params.timeRange?.to?.toISOString(),
+                    }
+                  : undefined,
+              ...column.aggConfig.serialize(),
```

Comment:
Is it possible for different columns to have different `appliedTimeRange`? It seems that way from the code

Reply:
not different (as this.params.timeRange doesn't change) but some columns will have it and some won't. also on the datatable level there could actually be dfferent timeranges on different columns, but that's not in the scope of tabify.